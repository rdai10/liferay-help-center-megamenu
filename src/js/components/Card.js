import preact from 'preact';
import PropTypes from 'prop-types';

const Card = ({description, name, onClick, svgId, type, url}) => {
	return (
		<div class={`${type}-card`}>
			<a class="autofit-row autofit-row-center" href={url} onClick={onClick}>
				{svgId && (
					<div class="autofit-col">
						<svg class="icon">
							<use xlinkHref={svgId} />
						</svg>
					</div>
				)}

				<div class="autofit-col autofit-col-expand">
					{name && (
						<h4 class="title">
							{name}
							{type === 'product' ? (
								<svg class="lexicon-icon lexicon-icon-arrow-right">
									<use xlinkHref="#arrow-right" />
								</svg>
							) : (
								''
							)}
						</h4>
					)}

					{description && (
						<p class="description">{description}</p>
					)}
				</div>
			</a>
		</div>
	);
};

Card.propTypes = {
	description: PropTypes.string,
	name: PropTypes.node,
	onClick: PropTypes.func,
	svgId: PropTypes.string,
	type: PropTypes.oneOf(['nav', 'product']).isRequired,
	url: PropTypes.string
};

export default Card;