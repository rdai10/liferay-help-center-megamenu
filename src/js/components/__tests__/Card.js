import preact from 'preact';
import {cleanup, fireEvent, render} from 'preact-testing-library';

import Card from '../Card';

afterEach(cleanup);

describe('Card', () => {
	it('renders nav card correctly', () => {
		const {container} = render(
			<Card
				description="Description"
				name="Name"
				svgId="#ticket"
				type="nav"
				url="/"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders product card correctly', () => {
		const {container} = render(
			<Card
				description="Description"
				name="Name"
				svgId="#ticket"
				type="product"
				url="/"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders without description', () => {
		const {container} = render(
			<Card name="Name" svgId="#ticket" type="nav" url="/" />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders without name', () => {
		const {container} = render(
			<Card
				description="Description"
				svgId="#ticket"
				type="nav"
				url="/"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders without svg', () => {
		const {container} = render(
			<Card name="Name" description="Description" type="nav" url="/" />
		);

		expect(container).toMatchSnapshot();
	});

	it('renders without url', () => {
		const {container} = render(
			<Card
				name="Name"
				description="Description"
				onClick={jest.fn()}
				svgId="#ticket"
				type="nav"
			/>
		);

		expect(container).toMatchSnapshot();
	});

	it('does something when clicked', () => {
		const handleOnClick = jest.fn();

		const {getByText} = render(
			<Card
				name="Name"
				description="Description"
				onClick={handleOnClick}
				svgId="#ticket"
				type="nav"
			/>
		);

		fireEvent.click(getByText('Name'));

		expect(handleOnClick).toBeCalledTimes(1);
	});
});