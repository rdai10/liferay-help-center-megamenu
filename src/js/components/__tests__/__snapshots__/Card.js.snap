// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders nav card correctly 1`] = `
<div>
  <div
    class="nav-card"
  >
    <a
      class="autofit-row autofit-row-center"
      href="/"
    >
      <div
        class="autofit-col"
      >
        <svg
          class="icon"
        >
          <use
            href="#ticket"
          />
        </svg>
      </div>
      <div
        class="autofit-col autofit-col-expand"
      >
        <h4
          class="title"
        >
          Name
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
    </a>
  </div>
</div>
`;

exports[`Card renders product card correctly 1`] = `
<div>
  <div
    class="product-card"
  >
    <a
      class="autofit-row autofit-row-center"
      href="/"
    >
      <div
        class="autofit-col"
      >
        <svg
          class="icon"
        >
          <use
            href="#ticket"
          />
        </svg>
      </div>
      <div
        class="autofit-col autofit-col-expand"
      >
        <h4
          class="title"
        >
          Name
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
    </a>
  </div>
</div>
`;

exports[`Card renders without description 1`] = `
<div>
  <div
    class="nav-card"
  >
    <a
      class="autofit-row autofit-row-center"
      href="/"
    >
      <div
        class="autofit-col"
      >
        <svg
          class="icon"
        >
          <use
            href="#ticket"
          />
        </svg>
      </div>
      <div
        class="autofit-col autofit-col-expand"
      >
        <h4
          class="title"
        >
          Name
        </h4>
        
      </div>
    </a>
  </div>
</div>
`;

exports[`Card renders without name 1`] = `
<div>
  <div
    class="nav-card"
  >
    <a
      class="autofit-row autofit-row-center"
      href="/"
    >
      <div
        class="autofit-col"
      >
        <svg
          class="icon"
        >
          <use
            href="#ticket"
          />
        </svg>
      </div>
      <div
        class="autofit-col autofit-col-expand"
      >
        
        <p
          class="description"
        >
          Description
        </p>
      </div>
    </a>
  </div>
</div>
`;

exports[`Card renders without svg 1`] = `
<div>
  <div
    class="nav-card"
  >
    <a
      class="autofit-row autofit-row-center"
      href="/"
    >
      
      <div
        class="autofit-col autofit-col-expand"
      >
        <h4
          class="title"
        >
          Name
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
    </a>
  </div>
</div>
`;

exports[`Card renders without url 1`] = `
<div>
  <div
    class="nav-card"
  >
    <a
      class="autofit-row autofit-row-center"
    >
      <div
        class="autofit-col"
      >
        <svg
          class="icon"
        >
          <use
            href="#ticket"
          />
        </svg>
      </div>
      <div
        class="autofit-col autofit-col-expand"
      >
        <h4
          class="title"
        >
          Name
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
    </a>
  </div>
</div>
`;
